### PR TITLE
In 139486493 show error messages

### PR DESCRIPTION
**Pull request by ThomasOSI**

### CODE REVIEW

File: lib/api_error.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      @api_error[:http_code] = api_response.status if api_response.status.present?
+    end
+
+    super(exception_info[:message])
+  end
+
+  def generate_base_message(exception_info)
+    {
+      response_body: 'N/A',
+      url: exception_info[:api_url],
```
Comment (NateOSI) [inline lib/api_error.rb]: exception_info.as_json(only: [:method, : sent_attributes, :api_url)

File: spec/controllers/api/v1/participants_controller_spec.rb [hunk 2]
```diff
@@ -27,19 +26,73 @@
...
+          'status': 'N/A',
+          'message': 'execution expired',
+          'api_response_body': 'N/A',
+          'method': 'post',
+          'url': '/api/v1/participants'
+        }
+
+        body_as_hash = JSON.parse(response.body)
+        expect(body_as_hash['sent_attributes']).to be_present
+        expect(body_as_hash.except('sent_attributes')).to eq(expected_exception.as_json)
```
Comment (NateOSI) [inline spec/controllers/api/v1/participants_controller_spec.rb]:
``` 
# use a_hash_including then you don't need to except 
# and it reads better as to what it actually returns

expect(body).to match a_hash_including(
             'error': 'api_error',
             'status': 'N/A',
             'message': 'execution expired',
             'api_response_body': 'N/A',
             'method': 'post',
             'url': '/api/v1/participants'
            )
```

File: spec/lib/intake_faraday_middleware_spec.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      response_env = env.merge(status: 500, body: 'kaboom')
+      response = Faraday::Response.new(response_env)
+
+      expect(app).to receive(:call)
+        .with(env)
+        .and_return(response)
+
+      expect do
+        middleware.call(env)
+      end.to raise_error(ApiError)
```
Comment (NateOSI) [inline spec/lib/intake_faraday_middleware_spec.rb]: You could check error message. Not the end of the world though

File: lib/api_error.rb (hunk 1 above)
Comment (ThomasOSI) [inline lib/api_error.rb]: Changed api_url to url and changed this bit as recommended

File: spec/lib/intake_faraday_middleware_spec.rb (hunk 3 above)
Comment (ThomasOSI) [inline spec/lib/intake_faraday_middleware_spec.rb]: done!

**Pull request merged**